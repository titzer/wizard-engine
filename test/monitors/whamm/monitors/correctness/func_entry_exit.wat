(module
  (func $putc (import "wizeng" "putc") (param i32))
  (func $puti (import "wizeng" "puti") (param i32))

  (func  $func_entry_probe (export "wasm:func:entry(fid,pc)") (param i32 i32)
    (call $putc (i32.const 70)) ;; 'F'
    (call $puti (local.get 0))
    (call $putc (i32.const 44)) ;; ','
    (call $puti (local.get 1))
    (call $putc (i32.const 10)) ;; '\n'
  )
  (func $func_exit_probe (export "wasm:func:exit(fid,pc)") (param i32 i32)
    (call $putc (i32.const 88)) ;; 'X'
    (call $puti (local.get 0))
    (call $putc (i32.const 44)) ;; ','
    (call $puti (local.get 1))
    (call $putc (i32.const 10)) ;; '\n'
  )
)
