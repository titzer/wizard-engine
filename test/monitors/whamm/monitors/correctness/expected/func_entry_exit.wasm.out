F0,0
F1,0
F1,0
F2,0
X2,30
F1,0
X1,384
X1,376
F1,0
F3,0
F4,0
F5,0
X5,4
F6,0
F7,0
X7,3
X6,4
X4,9
X3,3
X1,376
X1,376
X0,14
